### PR TITLE
chore: add analytics reference lines and formatting to saved chart ver…

### PR DESCRIPTION
**Pull request by rephus**
…sion

Closes: #4295

### Description:
Updating a chart with reference lines

Screenshot from 2023-01-27 17-11-29

Updating a table with cond formatting
Screenshot from 2023-01-27 17-11-06

image

### CODE REVIEW

**Review by TuringLovesDeathMetal — APPROVED** [review]
yay! looks great, thank you ❤️

File: packages/backend/src/services/SavedChartsService/SavedChartService.ts [hunk 1]
```diff
@@ -151,6 +171,10 @@ export class SavedChartService {
             properties: {
                 projectId: savedChart.projectUuid,
                 savedQueryId: savedChartUuid,
+                constantReferenceLinesNum: 0,
+                conditionalFormattingNumRules: 0,
+                showLegend: true,
+                margins: 'default',
```
Comment (ZeRego) [inline packages/backend/src/services/SavedChartsService/SavedChartService.ts:177]: Doesn't make sense to have these properties in this event. 🤔 We only need it in the version creation.

File: packages/backend/src/services/SavedChartsService/SavedChartService.ts [hunk 2]
```diff
@@ -72,6 +81,16 @@ export class SavedChartService {
                 savedChart.metricQuery.tableCalculations.length,
             pivotCount: (savedChart.pivotConfig?.columns || []).length,
             chartType: savedChart.chartConfig.type,
+            constantReferenceLinesNum: echartsConfig?.series?.filter(
```
Comment (ZeRego) [inline packages/backend/src/services/SavedChartsService/SavedChartService.ts:84]: I know Katie suggested these properties names but I would change them to match the existing ones. 
`referenceLinesCount` and `conditionalFormattingRulesCount`
What do you think ?

Comment (ZeRego) [reply packages/backend/src/services/SavedChartsService/SavedChartService.ts:84]: We also have a section for the cartesian metadata in line 94. Should we move that one in there ? 
and have a new one for the table metadata?

Comment (TuringLovesDeathMetal) [reply packages/backend/src/services/SavedChartsService/SavedChartService.ts:84]: yep, naming is great! consistency is our friend ❤️

Comment (rephus) [reply packages/backend/src/services/SavedChartsService/SavedChartService.ts]: Updated 
Screenshot from 2023-01-30 12-14-56

File: packages/backend/src/analytics/LightdashAnalytics.ts [hunk 3]
```diff
@@ -162,6 +162,12 @@ export type CreateSavedChartOrVersionEvent = BaseTrack & {
             yAxisCount: number;
             seriesCount: number;
             seriesTypes: CartesianSeriesType[];
+            referenceLinesCount?: number;
+            margins?: string;
+            showLegend?: boolean;
+        };
+        table?: {
+            conditionalFormattingRulesCount?: number;
```
Comment (ZeRego) [inline packages/backend/src/analytics/LightdashAnalytics.ts:170]: We should make these required so the count values default to 0 instead of undefined.

**Review by ZeRego — APPROVED** [review]

**Pull request merged**